encoding/template/run_length_decode.go

```diff
@@ -11,22 +11,22 @@ var (
 	errNonPositiveNumberOfRepetitions = errors.New("non positive number of repetitions")
 )
 
-// readValueFn reads a GenericValue from an `io.Reader`.
-type readValueFn func(reader io.Reader) (GenericValue, error)
+// readValueFn reads a GenericRLValue from an `io.Reader`.
```

Comment:
nit: `RLValue` -> `RLEValue`.

Reply:
Ah, just realized that I don't need to do this. I moved all the generic types to `generic.go`.